### PR TITLE
Add TensorFlow GPU Dockerfile

### PR DESCRIPTION
**Pull request by NathanDeMaria**
TensorFlow built from source, compiled to use instructions that can speed up CPU.

Built on `Miniconda`, using `python3.5`

Up on hub.docker.com as `hudl/tensorflow-gpu:latest`

### CODE REVIEW

File: tensorflow-gpu/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+ENV CUDA_ROOT /usr/local/cuda/bin
+RUN apt-get update -y && apt-get install git wget vim libgtk2.0-0 -y
+
+# Miniconda/basic packages install
+RUN wget --quiet https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O install_anaconda.sh && \
+    /bin/bash install_anaconda.sh -b -p /opt/conda && \
+    /opt/conda/bin/conda update conda && \
+    # Set default python to 3.5
+    /opt/conda/bin/conda install python=3.5 && \
+    rm install_anaconda.sh
```
Comment (rversaw) [inline tensorflow-gpu/Dockerfile]: Extremely minor, but should we rename this to `install_miniconda.sh` instead? Just to make it a tiny bit more clear.

Comment (NathanDeMaria) [reply tensorflow-gpu/Dockerfile]: Lol at my copy pasting. Good catch

**Review by jbencook — APPROVED** [review]
This looks good. Did it increase the size of the image?

**Comment by NathanDeMaria** [comment]
Actually dropped the size of the base image, I think because of the switch to `miniconda`. But our dev container is slightly larger now (4.872GB vs 4.826GB previously)

**Comment by rversaw** [comment]
Any idea where the space goes on this - is it leftover artifacts (checkout directories) or something else? It looks like most of the official GPU images are around 1GB for standard and 2GB for `devel`.

https://hub.docker.com/r/tensorflow/tensorflow/tags/

**Comment by NathanDeMaria** [comment]
This image is actually [~2GB (compressed)](https://hub.docker.com/r/hudl/tensorflow-gpu/tags/), ~3.8GB uncompressed. Our packages bring it up to the 4.872GB (uncompressed)

**Pull request merged**